File: src/main/java/jp/kentan/minecraft/neko_core/event/ZoneEvent.java
package jp.kentan.minecraft.neko_core.event;

import org.bukkit.block.Sign;
import org.bukkit.entity.Player;
import org.bukkit.event.block.SignChangeEvent;

public interface ZoneEvent {
    void onPlayerJoin(Player player);

    void onSignPlace(SignChangeEvent event);

    void onSignBreak(Player player, Sign sign);

    void onSignClick(Player player, Sign sign);
}

File: src/main/kotlin/jp/kentan/minecraft/nekocore/event/ZoneEvent.kt
package jp.kentan.minecraft.nekocore.event

import org.bukkit.block.Sign
import org.bukkit.entity.Player
import org.bukkit.event.block.SignChangeEvent

interface ZoneEvent {
    fun onPlayerJoin(player: Player)

    fun onSignPlace(event: SignChangeEvent)

    fun onSignBreak(player: Player, sign: Sign)

    fun onSignClick(player: Player, sign: Sign)
}